src/index.js

```diff
@@ -11,10 +14,42 @@ const MAX_WAIT_TIME_ON_EXIT = 30*1000;
 logger.debug('Current config: %o', config);
 
 let intervalId;
-if (!isInDebugMode()) {
-    intervalId = setInterval(run, config.refreshInterval);
+
+if (config.concatEventPolicy === "true") {
+    let state = await getState(compactor.getOpts(), compactor.getMinioClient());
+
+    if(state.size === 0) {
+        try {
+            // Run the compact function immediately at launch
+            logger.info('Running compactor at launch...');
+            await run();
+            logger.info('Compactor initialized.');
+        } catch (error) {
+            logger.error('Error during compactor initialization:', error);
+        }
+    }
+
+    // Start consuming messages
+    (async () => {
+        await compactor.startKafkaConsumer();
+    })();
 } else {
-    run();
+    try {
+        // Run the compact function immediately at launch
+        logger.info('Running compactor at launch...');
+        await run();
+
+        // Set an interval to run compact periodically if not in debug mode
+        if (!isInDebugMode()) {
+            intervalId = setInterval(run, config.refreshInterval);
+        } else {
+            await run();
+        }
+
+        logger.info('Compactor initialized.');
+    } catch (error) {
+        logger.error('Error during compactor initialization:', error);
+    }
 }
```

Comment:
Define two functions with the actual code and call them here

Reply:
fixed